src/Utils.js

```diff
@@ -189,7 +189,7 @@ export function selectorError(selector, type = '') {
   );
 }
 
-export const isCompoundSelector = /([a-z]\.[a-z]|[a-z]\[.*\]|[a-z]#[a-z])/i;
+export const isCompoundSelector = /^[\.#]?-?[_a-z]+[_a-z0-9-]*[\.\[#]/i;
```

Comment:
`[\.#]?          # anything but a hash, 0 or 1`
not sure it matters here but I think this one is actually '. or a #, 0 or 1'